Improve integration tests

C0urante opened the pull request:
As [raised in #214](https://github.com/aiven/jdbc-connector-for-apache-kafka/pull/214#issuecomment-1457060161), the integration tests for this project no longer run correctly. This PR originally fixed those integration tests, but now depends on https://github.com/aiven/jdbc-connector-for-apache-kafka/pull/224 for that bit.

This PR:
- Tweaks the build file to allow integration tests to be run without also running unit tests
- Tweaks the build file to enable logging for integration tests

The first change can be verified locally by running `./gradlew clean build integrationTest`. I only see effects from the second change while running in IntelliJ; this still seems valuable, but if not, I can revert.

jeqo commented on build.gradle:
```diff
@@ -215,7 +209,7 @@ task integrationTest(type: Test) {
     classpath = sourceSets.integrationTest.runtimeClasspath
 
-    dependsOn test, distTar
+    dependsOn testClasses, distTar
```
I'm probably lacking some context, but why would we need to avoid running test before integration tests? 
If we really want to, we could use `./gradlew integrationTest -x test`.

Also, would like to understand more the dependency to distTar for integration tests.

First time configuring IT on gradle, so looking at https://docs.gradle.org/7.3/userguide/java_testing.html#sec:configuring_java_integration_tests
`shouldRunAfter test` may be a more idiomatic way of defining this? 

Also, we could use `check.dependsOn integrationTest` to link ITs to gradle check, but we can do that on a separate PR to tackle #221

C0urante replied:
Requiring people to explicitly opt out of running unit tests before running integration tests seems less friendly and was a bit of a headache personally when writing new integration tests for https://github.com/aiven/jdbc-connector-for-apache-kafka/pull/223. This is similar to how Kafka does it, although they go one step further and define three different Gradle goals: `unitTest`, `integrationTest`, and `test` (which runs both unit and integration tests).

RE the dependency on `distTar`: I'm honestly not sure about this one, but chose to leave it in for now as it doesn't appear to be causing any issues.

jeqo replied:
Got it, that make sense. Then, for connectors: `test` is for unit-tests, and `integrationTest` for ITs and runs independently from unit-tests.

May be worth adding a note to clarify this?
```suggestion
    // ITs run independent from unit-tests
    dependsOn testClasses, distTar
```
 
Also, could `gradle check` be the way to both tests, or `test integrationTest` should suffice?

C0urante replied:
👍 for adding the comment

I tried attaching the `integrationTest` task to `check` but the end result was that `./gradlew build -x test` ended up running integration tests (since we didn't exclude that task explicitly). So at least with the latest version of the PR, you have to run `test integrationTest` instead of just `check` if you'd like to run both categories of tests.

jeqo replied:
oh I see. That make sense. Thanks @C0urante

AnatolyPopov reviewed (changes requested):
Left some minor comments and also created a [PR](https://github.com/aiven/jdbc-connector-for-apache-kafka/pull/224) to enable integration tests in CI + minimum changes to fix the tests there. Could you rebase on top of it? Some changes from this PR still make sense despite the fact that tests are fixed with another one.

C0urante commented:
Thanks @AnatolyPopov, I've rebased on https://github.com/aiven/jdbc-connector-for-apache-kafka/pull/224 which I believe addresses your comments regarding dependency versions.

In the future, I think we may want to reconsider how we choose versions for these dependencies. Right now they're only used in integration tests and one of the purposes they serve is to run our embedded Kafka Connect worker. Ideally we should make sure that the versions we choose align with whatever version of Kafka Connect we're testing against (which can be seen [here](https://github.com/apache/kafka/blob/2.2.0/build.gradle#L1461-L1478) and [here](https://github.com/apache/kafka/blob/2.2.0/gradle/dependencies.gradle#L49-L155)); otherwise, the scenario we end up testing against is a bit unrealistic. But this obviously doesn't have to happen now and may become even less necessary if we end up adopting the [embedded testing framework](https://github.com/apache/kafka/blob/5f6a050bfee09b634497f9ba35e2964289be1e4d/connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java) that Kafka Connect uses these days instead of our own one-off solution.

AnatolyPopov commented:
@C0urante Thanks! And you comment definitely makes sense. We perhaps need to look into embedded testing framework also, thanks for the link.

jeqo reviewed (approved):
LGTM!

AnatolyPopov reviewed (approved):
LGTM
